libs/framework/src/bundle_archive.c

```diff
@@ -28,644 +28,415 @@
+    char manifestPathBuffer[CELIX_DEFAULT_STRING_CREATE_BUFFER_SIZE];
+    char* manifestPath = celix_utils_writeOrCreateString(manifestPathBuffer, sizeof(manifestPathBuffer), "%s/%s", archive->resourceCacheRoot, CELIX_BUNDLE_MANIFEST_REL_PATH);
+    if (celix_utils_fileExists(manifestPath)) {
+        status = celix_utils_getLastModified(manifestPath, lastModified);
+    } else {
+        lastModified->tv_sec = 0;
+        lastModified->tv_nsec = 0;
+    }
+    celix_utils_freeStringIfNotEqual(manifestPathBuffer, manifestPath);
+    return status;
 }
 
-static celix_status_t bundleArchive_readLastModified(bundle_archive_pt archive, time_t *time) {
-	FILE *lastModifiedFile;
-	char lastModified[512];
-
-	celix_status_t status = CELIX_SUCCESS;
-
-	snprintf(lastModified, sizeof(lastModified), "%s/bundle.lastmodified", archive->archiveRoot);
-
-	lastModifiedFile = fopen(lastModified, "r");
-	if (lastModifiedFile == NULL) {
-		status = CELIX_FILE_IO_EXCEPTION;
-	} else {
-		char timeStr[20];
-		int year, month, day, hours, minutes, seconds;
-		struct tm tm_time;
-		memset(&tm_time,0,sizeof(struct tm));
-
-		if (fgets(timeStr, sizeof(timeStr), lastModifiedFile) == NULL) {
-			status = CELIX_FILE_IO_EXCEPTION;
-		}
-		fclose(lastModifiedFile);
-		if (status == CELIX_SUCCESS) {
-			sscanf(timeStr, "%d %d %d %d:%d:%d", &year, &month, &day, &hours, &minutes, &seconds);
-			tm_time.tm_year = year - 1900;
-			tm_time.tm_mon = month - 1;
-			tm_time.tm_mday = day;
-			tm_time.tm_hour = hours;
-			tm_time.tm_min = minutes;
-			tm_time.tm_sec = seconds;
-
-			*time = mktime(&tm_time);
-		}
-	}
-
-	framework_logIfError(archive->fw->logger, status, NULL, "Could not read last modified");
-
-	return status;
+celix_status_t celix_bundleArchive_getLastModified(bundle_archive_pt archive, struct timespec* lastModified) {
+    celix_status_t status;
+    celixThreadMutex_lock(&archive->lock);
+    status = celix_bundleArchive_getLastModifiedInternal(archive, lastModified);
+    celixThreadMutex_unlock(&archive->lock);
+    return status;
 }
 
-static celix_status_t bundleArchive_writeLastModified(bundle_archive_pt archive) {
-	celix_status_t status = CELIX_SUCCESS;
-	FILE *lastModifiedFile;
-	char lastModified[512];
-
-	snprintf(lastModified, sizeof(lastModified), "%s/bundle.lastmodified", archive->archiveRoot);
-	lastModifiedFile = fopen(lastModified, "w");
-	if (lastModifiedFile == NULL) {
-		status = CELIX_FILE_IO_EXCEPTION;
-	} else {
-		char timeStr[20];
-		strftime(timeStr, 20, "%Y %m %d %H:%M:%S", localtime(&archive->lastModified));
-		fprintf(lastModifiedFile, "%s", timeStr);
-		fclose(lastModifiedFile);
-	}
-
-	framework_logIfError(celix_frameworkLogger_globalLogger(), status, NULL, "Could not write last modified");
-
-	return status;
+celix_status_t bundleArchive_setLastModified(bundle_archive_pt archive __attribute__((unused)), time_t lastModifiedTime  __attribute__((unused))) {
+    return celix_utils_touch(archive->archiveRoot);
```

Comment:
It does not feel right to change root's timestamp while getting manifest's timestamp.